File: app/src/main/java/org/wikiedufoundation/wikiedudashboard/course_detail/articles_edited/view/CourseArticlesEditedFragment.java
package org.wikiedufoundation.wikiedudashboard.course_detail.articles_edited.view;

import android.content.Context;
import android.os.Bundle;
import android.support.v4.app.Fragment;
import android.support.v7.widget.LinearLayoutManager;
import android.support.v7.widget.RecyclerView;
import android.util.Log;
import android.view.LayoutInflater;
import android.view.View;
import android.view.ViewGroup;
import android.widget.ProgressBar;
import android.widget.TextView;

import org.wikiedufoundation.wikiedudashboard.R;
import org.wikiedufoundation.wikiedudashboard.course_detail.articles_edited.data.ArticlesEdited;

import org.wikiedufoundation.wikiedudashboard.course_detail.articles_edited.presenter.ArticlesEditedPresenterImpl;
import org.wikiedufoundation.wikiedudashboard.course_detail.articles_edited.provider.RetrofitArticlesEditedProvider;
import org.wikiedufoundation.wikiedudashboard.helper.SharedPrefs;
import org.wikiedufoundation.wikiedudashboard.helper.ViewUtils;

import butterknife.BindView;
import butterknife.ButterKnife;

public class CourseArticlesEditedFragment extends Fragment implements ArticlesEditedView {


    private SharedPrefs sharedPrefs;
    private Context context;
    @BindView(R.id.tv_no_edited_articles)
    TextView tv_no_edited_articles;



    @BindView(R.id.progressBar)
    ProgressBar progressBar;

    @BindView(R.id.rv_edited_articles_list)
    RecyclerView recyclerView;

    private ArticlesEditedPresenterImpl articlesEditedPresenter;
    private ArticlesEditedRecyclerAdapter articlesEditedRecyclerAdapter;
    private String url;


    public CourseArticlesEditedFragment(){

    }

    @Override
    public View onCreateView(LayoutInflater inflater, ViewGroup container, Bundle savedInstanceState) {
        View view = inflater.inflate(R.layout.fragment_articles_edited,container,false);
        ButterKnife.bind(this,view);
        url = getArguments().getString("url",null);
        context = getContext();
        sharedPrefs = new SharedPrefs(context);
        tv_no_edited_articles.setText(sharedPrefs.getCookies());
        articlesEditedPresenter = new ArticlesEditedPresenterImpl(new RetrofitArticlesEditedProvider(),this);

        articlesEditedRecyclerAdapter = new ArticlesEditedRecyclerAdapter(context);
        LinearLayoutManager linearLayoutManager = new LinearLayoutManager(context);
        recyclerView.setLayoutManager(linearLayoutManager);
        recyclerView.setHasFixedSize(true);
        recyclerView.setAdapter(articlesEditedRecyclerAdapter);

        articlesEditedPresenter.requestArticlesEdited(url);
        return view;
    }

    @Override
    public void setData(ArticlesEdited data) {
        Log.d("ArticlesEditedFragment",data.toString());
        if(data.getCourse().getArticles().size()>0)
        {
            recyclerView.setVisibility(View.VISIBLE);
            articlesEditedRecyclerAdapter.setData(data.getCourse().getArticles());
            articlesEditedRecyclerAdapter.notifyDataSetChanged();
            tv_no_edited_articles.setVisibility(View.GONE);

        }
        else {
            recyclerView.setVisibility(View.GONE);
            tv_no_edited_articles.setVisibility(View.VISIBLE);
        }
    }
    @Override
    public void showProgressBar(boolean show) {
        if (show){
            progressBar.setVisibility(View.VISIBLE);
        }else {
            progressBar.setVisibility(View.GONE);
        }
    }

    @Override
    public void showMessage(String message) {
        ViewUtils.showToast(context, message);
    }

}

File: app/src/main/java/org/wikiedufoundation/wikiedudashboard/ui/coursedetail/articlesedited/view/CourseArticlesEditedFragment.kt
package org.wikiedufoundation.wikiedudashboard.ui.coursedetail.articlesedited.view

import android.os.Bundle
import android.view.LayoutInflater
import android.view.View
import android.view.ViewGroup
import androidx.fragment.app.Fragment
import androidx.recyclerview.widget.LinearLayoutManager
import kotlinx.android.synthetic.main.fragment_articles_edited.*
import org.koin.android.ext.android.inject
import org.koin.core.parameter.parametersOf
import org.wikiedufoundation.wikiedudashboard.R
import org.wikiedufoundation.wikiedudashboard.ui.adapters.ArticlesEditedRecyclerAdapter
import org.wikiedufoundation.wikiedudashboard.ui.coursedetail.articlesedited.data.ArticlesEdited
import org.wikiedufoundation.wikiedudashboard.ui.coursedetail.articlesedited.presenter.ArticlesEditedPresenter
import org.wikiedufoundation.wikiedudashboard.ui.coursedetail.articlesedited.provider.RetrofitArticlesEditedProvider
import org.wikiedufoundation.wikiedudashboard.util.showSnackbar
import timber.log.Timber

/**
 * A [Fragment] subclass for edited course articles
 * ***/
class CourseArticlesEditedFragment : Fragment(), ArticlesEditedView {

    private val retrofitArticlesEditedProvider: RetrofitArticlesEditedProvider by inject()
    private val articlesEditedPresenter: ArticlesEditedPresenter by inject {
        parametersOf(this, retrofitArticlesEditedProvider)
    }

    private lateinit var articlesEditedRecyclerAdapter: ArticlesEditedRecyclerAdapter

    private var url: String? = null

    override fun onCreateView(
        inflater: LayoutInflater,
        container: ViewGroup?,
        savedInstanceState: Bundle?
    ): View? = inflater.inflate(R.layout.fragment_articles_edited, container, false)

    override fun onViewCreated(view: View, savedInstanceState: Bundle?) {
        url = arguments?.getString("url", null)

        articlesEditedRecyclerAdapter = ArticlesEditedRecyclerAdapter(R.layout.item_rv_articles_edited)

        recyclerEditedArticlesList.apply {
            layoutManager = LinearLayoutManager(context)
            setHasFixedSize(true)
            adapter = articlesEditedRecyclerAdapter
        }

        url?.let { articlesEditedPresenter.requestArticlesEdited(it) }
    }

    override fun setData(data: ArticlesEdited) {
        Timber.d(data.toString())
        if (data.course.articles.isNotEmpty()) {
            recyclerEditedArticlesList?.visibility = View.VISIBLE
            articlesEditedRecyclerAdapter.setData(data.course.articles)
            articlesEditedRecyclerAdapter.notifyDataSetChanged()
            textViewNoEditedArticles?.visibility = View.GONE
        } else {
            recyclerEditedArticlesList?.visibility = View.GONE
            textViewNoEditedArticles?.visibility = View.VISIBLE
        }
    }

    override fun showProgressBar(show: Boolean) {
        progressBar?.visibility = if (show) {
            View.VISIBLE
        } else {
            View.GONE
        }
    }

    override fun showMessage(message: String) {
        view?.showSnackbar(message)
    }
}
